plugin/src/main/java/com/h3xstream/findsecbugs/serial/UnsafeJacksonDeserializationDetector.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Find Security Bugs
+ * Copyright (c) Philippe Arteau, All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3.0 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.
+ */
+package com.h3xstream.findsecbugs.serial;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.FieldAnnotation;
+import edu.umd.cs.findbugs.ba.CFG;
+import edu.umd.cs.findbugs.ba.CFGBuilderException;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.ba.DataflowAnalysisException;
+import edu.umd.cs.findbugs.ba.Location;
+import org.apache.bcel.classfile.AnnotationEntry;
+import org.apache.bcel.classfile.ElementValuePair;
+import org.apache.bcel.classfile.Field;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.generic.ConstantPoolGen;
+import org.apache.bcel.generic.Instruction;
+import org.apache.bcel.generic.InvokeInstruction;
+import org.apache.bcel.generic.MethodGen;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Detect unsafe Jackson datatype deserialization
+ *
+ */
+public class UnsafeJacksonDeserializationDetector implements Detector {
+
+    private static final String DESERIALIZATION_TYPE = "JACKSON_UNSAFE_DESERIALIZATION";
+
+    private BugReporter bugReporter;
+
+
+    public UnsafeJacksonDeserializationDetector(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+    private static final List<String> ANNOTATION_TYPES = Arrays.asList(
+            "Lcom/fasterxml/jackson/annotation/JsonTypeInfo;");
+
+    private static final List<String> VULNERABLE_USE_NAMES = Arrays.asList(
+            "CLASS", "MINIMAL_CLASS");
+
+    private static final List<String> OBJECT_MAPPER_CLASSES = Arrays.asList(
+            "com.fasterxml.jackson.databind.ObjectMapper",
+            "org.codehaus.jackson.map.ObjectMapper");
+
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        JavaClass javaClass = classContext.getJavaClass();
+        if (OBJECT_MAPPER_CLASSES.contains(javaClass.getClassName())) {
```

Comment:
Correct. This is to avoid marking the jackson classes themselves as false positives.